File: src/data/courses/dsa/queues/queue-core-operations/implement-queue-operations/code/python/implement_queue_operations.py
class Queue:
    def __init__(self, capacity):
        self.arr = [0] * capacity
        self.front = 0
        self.rear = -1
        self.size = 0
    
    def enqueue(self, x):
        self.rear += 1
        self.arr[self.rear] = x
        self.size += 1
    
    def dequeue(self):
        self.size -= 1
        val = self.arr[self.front]
        self.front += 1
        return val
    
    def peek(self):
        return self.arr[self.front]
    
    def is_empty(self):
        return self.size == 0

File: src/data/courses/dsa/queues/queue-core-operations/implement-queue-operations/code/java/ImplementQueueOperations.java
class Queue {
    int[] arr;
    int front, rear, size;
    
    Queue(int capacity) {
        arr = new int[capacity];
        front = 0;
        rear = -1;
        size = 0;
    }
    
    void enqueue(int x) {
        arr[++rear] = x;
        size++;
    }
    
    int dequeue() {
        size--;
        return arr[front++];
    }
    
    int peek() {
        return arr[front];
    }
    
    boolean isEmpty() {
        return size == 0;
    }
}